packages/block-editor/src/hooks/border-color.js

```diff
@@ -48,8 +49,17 @@ export function BorderColorEdit( props ) {
 	const colors = useSetting( 'color.palette' ) || EMPTY_ARRAY;
 	const disableCustomColors = ! useSetting( 'color.custom' );
 	const disableCustomGradients = ! useSetting( 'color.customGradient' );
+	const [ colorValue, setColorValue ] = useState(
+		getColorObjectByAttributeValues(
+			colors,
+			borderColor,
+			style?.border?.color
+		)?.color
```

Comment:
Can we use lazy initialization here?

Reply:
Oh, that's a good one, thanks for catching that!